clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/ResizeGoogleServerGroupAtomicOperation.groovy

```diff
@@ -55,14 +55,26 @@ class ResizeGoogleServerGroupAtomicOperation implements AtomicOperation<Void> {
     def region = description.region
     def serverGroupName = description.serverGroupName
     def serverGroup = GCEUtil.queryServerGroup(googleClusterProvider, accountName, region, serverGroupName)
+    def isRegional = serverGroup.regional
+    // Will return null if this is a zonal server group.
     def zone = serverGroup.zone
     int targetSize = description.targetSize instanceof Number ? description.targetSize : description.capacity.desired
-    def instanceGroupManagers = compute.instanceGroupManagers()
 
-    instanceGroupManagers.resize(project,
-                                 zone,
-                                 serverGroupName,
-                                 targetSize).execute()
+    if (isRegional) {
+      def instanceGroupManagers = compute.regionInstanceGroupManagers()
+
+      instanceGroupManagers.resize(project,
+                                   region,
+                                   serverGroupName,
+                                   targetSize).execute()
+    } else {
+      def instanceGroupManagers = compute.instanceGroupManagers()
+
+      instanceGroupManagers.resize(project,
```

Comment:
I don't know how you feel about this, but it seems like a lot of this could be cleaned up with something like

```
def serverGroupLocation(serverGroup) {
  return serverGroup.regional ? serverGroup.region : serverGroup.zone
}

instanceGroupManagers.resize(project, serverGroupLocation(serverGroup), serverGroupName, targetSize)

```